veps/sig-compute/decentralized-migration/decentralized-migration-vep.md

```diff
@@ -0,0 +1,279 @@
+- Node label selectors for source virt-launcher pod. The node the source virt-launcher pod is running has labels that can be used for selecting which nodes the target can run on.
+- Source and Target `VirtualMachineInstance` UIDs. During the migration process, some assumptions have been made about the UID of the `VirtualMachineInstance`. Because it is now possible to migrate to a new `VirtualMachineInstance` the new and old UIDs have to communicated to the other side of the migration.
+- The new name of the `VirtualMachineInstance` and namespace. The domain xml name is generated from the `VirtualMachineInstace` name and namespace, and when migrating from the old domain to the new domain the new domain name has to be known at the source.
+
+For example:
+```yaml
+  status:
+    activePods:
+    ...
+   sourceMigrationState:
+      migrationConfiguration:
+        allowAutoConverge: false
+        allowPostCopy: false
+        allowWorkloadDisruption: false
+        bandwidthPerMigration: "0"
+        completionTimeoutPerGiB: 150
+        nodeDrainTaintKey: kubevirt.io/drain
+        parallelMigrationsPerCluster: 5
+        parallelOutboundMigrationsPerNode: 2
+        progressTimeout: 150
+        unsafeMigrationOverride: false
+      migrationNetworkType: Pod
+      mode: PreCopy
+      migrationUid: <uid>
+      node: <node>
+      pod: <virt-launcher-pod name>
+      vmiUID: <virtual machine instance UID>
+      nodeLabelSelectors: <map of key/value pairs>
+      namespace: <namespace of virtual machine instance>
+   targetMigrationState:
+      migrationConfiguration:
+        allowAutoConverge: false
+        allowPostCopy: false
+        allowWorkloadDisruption: false
+        bandwidthPerMigration: "0"
+        completionTimeoutPerGiB: 150
+        nodeDrainTaintKey: kubevirt.io/drain
+        parallelMigrationsPerCluster: 5
+        parallelOutboundMigrationsPerNode: 2
+        progressTimeout: 150
+        unsafeMigrationOverride: false
+      migrationNetworkType: Pod
+      mode: PreCopy
+      migrationUid: <uid>
+      node: <node>
+      directMigrationNodePorts:
+        "35665": 0
+        "35793": 49153
+        "42767": 49152
+      migrationUid: <uid>
+      node: <node>
+      nodeAddress: <address>
+      nodeDomainDetected: true
+      nodeDomainReadyTimestamp: "2025-02-19T19:02:15Z"
+      pod: <virt-launcher-pod name>
+      vmiUID: <virtual machine instance UID>
+      namespace: <namespace of virtual machine instance>
+      syncAddress: <address to connect for sync channel>
+```
+
+To keep backward compatibility if there is no `key`, it can be assumed it is both and it can follow both source and target flows in the `virt-controller`
+
+## Alternatives
+
+Completely rework the mechanism used for live migrations to be decentralized from the start. As there are quite a few issues with the current mechanism. This was rejected due to the amount of time and effort it would take to do a complete rewrite.
```

Comment:
Is there a well defined alternative approach? 
If such an approach exists perhaps we should at least describe it here briefly.

You could also mention here why some of the components, for example, the synchronizer live outside of core kubevirt.

Reply:
I am not aware of a well defined alternative approach. Besides maybe a complete redesign of the entire migration mechanism. But doing that alone is going to be a large undertaking. I can definitely explain the reasoning for having a separate migration controller vs having it part of lets say the virt-handler.